src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -276,86 +279,87 @@ internal void Authenticate()
             }
         }
 
-        internal void SendAsynchronously(HomeWorkspaceModel workspace, WorkspaceProperties workspaceProperties = null)
-        {
-            State = UploadState.Uploading;
-
-            Task.Factory.StartNew(() =>
-                {
-                    var result = this.Send(workspace, workspaceProperties);
-                    var serverResponce = serverResponceRegex.Match(result);
-
-                    if (serverResponce.Success)
-                    {
-                        State = UploadState.Succeeded;
-                        Error = UploadErrorType.None;
-                        CustomizerURL = String.Concat(serverUrl, serverResponce.Value);
-                    }
-                    else if (InvalidNodeNames != null)
-                    {
-                        Error = UploadErrorType.InvalidNodes;
-                    }
-                    else
-                    {
-                        // If there wasn't any error during uploading, 
-                        // that means it's some error on the server side.
-                        Error = UploadErrorType.UnknownServerError;
-                    }
-                });
-
-        }
-
-        /// <summary>
-        /// Sends workspace and its' dependencies to Flood.
-        /// </summary>
-        /// <returns>String which is response from server.</returns>
-        internal string Send(HomeWorkspaceModel workspace, WorkspaceProperties workspaceProperties = null)
+        internal async void SendAsync(HomeWorkspaceModel workspace, WorkspaceProperties workspaceProperties = null)
         {
             if (String.IsNullOrWhiteSpace(serverUrl))
             {
                 Error = UploadErrorType.ServerNotFound;
-                return Resources.FailedMessage;
+                return;
             }
 
             if (String.IsNullOrWhiteSpace(authenticationProvider.Username))
             {
                 Error = UploadErrorType.AuthenticationFailed;
-                return Resources.FailedMessage;
+                return;
             }
 
-            if (authenticationProvider == null)
+            if (workspace.Nodes.Count() == 0)
             {
-                Error = UploadErrorType.AuthProviderNotFound;
-                return Resources.FailedMessage;
+                Error = UploadErrorType.EmptyWorkspace;
+                return;
```

Comment:
This is the change for http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8875.